webapp/src/store/converters.ts

```diff
@@ -0,0 +1,35 @@
+import { DocumentData } from "firebase/firestore";
+import { Slot, Event } from "../util/types";
+
+
+// Any is used here because we literally aren't sure of the shape that's stored in the db.
+// To address this, we will need to use a converter on the doc where we have our snapshot coming in
+// https://firebase.google.com/docs/reference/node/firebase.firestore.FirestoreDataConverter
+// Issue #59
+
+export const docToEvent = (doc: DocumentData) => {
+    const data = doc.data();
+    if (data) {
+      return {
+        ...docToEventRaw(data),
+        id: doc.ref.id,
+      } as Event;
+    }
+    return null;
+  }
+  
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const docToEventRaw = (data: any) => {
```

Comment:
To be honest, I have a lot of gaps in my knowledge about how Firestore works, so I'm not sure if I'll be able to give much useful input in this area. But I guess it makes sense? I think we're using objects from firestore, converting whatever date representation they're using (maybe date strings?) to normal Date objects, and repackaging everything back up.

Reply:
Correct ^^.  Firestore has it's own timestamp object for everything, but it's nice to swap them to JavaScript dates in my opinion.  That's really the only thing we have to account for when swapping between firebase types and types for our app